File: examples/openmp/vec_add_sections.c
/* sections Directive Example */
#include <omp.h>
#define N 1000

main(int argc, char *argv[]) {

  int i;
  float a[N], b[N], c[N], d[N];

  /* Some initializations */
  for (i=0; i < N; i++) {
    a[i] = i * 1.5;
    b[i] = i + 22.35;
  }

  #pragma omp parallel shared(a,b,c,d) private(i)
  {

    #pragma omp sections nowait
    {

      #pragma omp section
      for (i=0; i < N; i++)
        c[i] = a[i] + b[i];

      #pragma omp section
      for (i=0; i < N; i++)
        d[i] = a[i] * b[i];

    }  /* end of sections */

  }  /* end of parallel region */

}

File: examples/openmp/vec_add_sections.f
!     SECTIONS Directive Example
      PROGRAM VEC_ADD_SECTIONS

      INTEGER N, I
      PARAMETER (N=1000)
      REAL A(N), B(N), C(N), D(N)

!     Some initializations
      DO I = 1, N
        A(I) = I * 1.5
        B(I) = I + 22.35
      ENDDO

!$OMP PARALLEL SHARED(A,B,C,D), PRIVATE(I)

!$OMP SECTIONS

!$OMP SECTION
      DO I = 1, N
        C(I) = A(I) + B(I)
      ENDDO

!$OMP SECTION
      DO I = 1, N
        D(I) = A(I) * B(I)
      ENDDO

!$OMP END SECTIONS NOWAIT

!$OMP END PARALLEL

      END
